seqr/utils/elasticsearch/es_search.py

```diff
@@ -318,27 +317,26 @@ def _filter_by_genotype(self, inheritance_mode, inheritance_filter, quality_filt
                             sample_ids += [
                                 sample_id for sample_id, sample in samples_by_id.items()
                                 if self._family_individual_affected_status[family_guid][sample.individual.guid] == Individual.AFFECTED_STATUS_AFFECTED]
-                        genotypes_q = _any_affected_sample_filter(sample_ids)
+                        self._index_searches[index].append(self._search.filter(_any_affected_sample_filter(sample_ids)))
                         self._any_affected_sample_filters = True
                     else:
                         # If searching across all families in an index with no inheritance mode we do not need to explicitly
                         # filter on inheritance, as all variants have some inheritance for at least one family
                         self._no_sample_filters = True
                         no_filter_indices.add(index)
-                        continue
+                    continue
 
-            if not genotypes_q:
-                for family_guid in sorted(family_samples_by_id.keys()):
-                    family_samples_q = self._get_family_sample_query(
-                        family_guid, family_samples_by_id, quality_filters_by_family,
-                        index_fields, inheritance_mode, inheritance_filter
-                    )
-                    if not genotypes_q:
-                        genotypes_q = family_samples_q
-                    else:
-                        genotypes_q |= family_samples_q
+            genotypes_qs = [
+                self._get_family_sample_query(
+                    family_guid, family_samples_by_id, quality_filters_by_family,
+                    index_fields, inheritance_mode, inheritance_filter
+                ) for family_guid in sorted(family_samples_by_id.keys())
+            ]
+            if len(genotypes_qs) > MAX_SEARCH_CLAUSES:
+                self._index_searches[index].append(self._search.filter(_or_filters(genotypes_qs[:MAX_SEARCH_CLAUSES])))
+                genotypes_qs = genotypes_qs[MAX_SEARCH_CLAUSES:]
```

Comment:
What happens if len(genotypes_qs) > 2*MAX_SEARCH_CLAUSES?

Reply:
we will already have migrated off of elasticsearch. The largest project we have is like 1600 families and it is currently not getting any new samples, the next largest we have is about 1000 and at the rate we get new families with data, it will be like 3 years before we even potentially get close to 2*1024=2048